lib/SILOptimizer/Utils/ConstantFolding.cpp

```diff
@@ -1316,6 +1316,42 @@ case BuiltinValueKind::id:
   return nullptr;
 }
 
+static void diagnosePotentialOutOfBoundsArrayLiteralSubscript(ApplyInst *AI) {
+  if (AI->getNumOperands() > 0 && isa<FunctionRefInst>(AI->getOperand(0))) {
```

Comment:
The TupleExtract, needs to take a 1 I think to do the pattern match.

Reply:
`m_SemanticsCallee` doesn't work for me because it fails due to `implicit instantiation of undefined template`. Maybe you're using a different version of the semantics matcher than I am?

Reply:
What is the error you are seeing?

Reply:
```
Implicit instantiation of undefined template 'swift::PatternMatch::Callee_match<swift::PatternMatch::Callee_match<swift::PatternMatch::Semantics_exactmatch> >'

2. In instantiation of template class 'swift::PatternMatch::match_combine_and<swift::PatternMatch::Callee_match<swift::PatternMatch::Callee_match<swift::PatternMatch::Semantics_exactmatch> >, swift::PatternMatch::Argument_match<swift::PatternMatch::bind_ty<swift::IntegerLiteralInst> > >' requested here
```

```
Implicit instantiation of undefined template 'swift::PatternMatch::Callee_match<swift::PatternMatch::Callee_match<swift::PatternMatch::Semantics_exactmatch> >'

2. In instantiation of function template specialization 'swift::PatternMatch::m_ApplyInst<0, swift::PatternMatch::Callee_match<swift::PatternMatch::Semantics_exactmatch>, swift::PatternMatch::bind_ty<swift::IntegerLiteralInst> >' requested here
```

two more like this.

Also, `m_StructInst(m_IntegerLiteralInst(index))` doesn't work (gives `No matching function for call to 'm_StructInst'`).

Reply:
Actually, the problem here is that I never implemented m_Struct (*doh*). I have a patch I am hacking on to add support for it. If you beat me to it, push it up here. Otherwise, I will finish it later today/tomorrow (it is using a little bit of variadic goop).

Reply:
I think I'll just wait for you to finish it as I am not familiar with pattern matching code :)

Reply:
Any updates on your patch @gottesmm? Thanks!